File: scripts/seed-quest-mosques.ts
import { readFileSync } from "node:fs";
import { resolve } from "node:path";
import { db } from "@/db";
import type { NewQuestMosque } from "@/db/schema";
import { questMosques } from "@/db/schema";

type JaisMosque = {
	jaisId: number;
	nama: string;
	alamat: string;
	daerah: string;
};

const raw: JaisMosque[] = JSON.parse(
	readFileSync(resolve(__dirname, "../data/jais-petaling.json"), "utf-8"),
);

const mosques: NewQuestMosque[] = raw.map((m) => ({
	name: m.nama,
	address: m.alamat,
	district: m.daerah,
	jaisId: String(m.jaisId),
}));

async function seed() {
	console.log(`Seeding ${mosques.length} quest mosques...`);

	await db
		.insert(questMosques)
		.values(mosques)
		.onConflictDoNothing({ target: questMosques.jaisId });

	console.log("Seeding complete.");
}

seed()
	.then(() => {
		console.log("Done!");
		process.exit(0);
	})
	.catch((err) => {
		console.error("Seed failed:", err);
		process.exit(1);
	});

File: src/bin/seed_quest_mosques.rs
use sedekahje::db;
use serde::Deserialize;

#[derive(Deserialize)]
struct JaisRecord {
    jaisId: i64,
    nama: String,
    alamat: String,
    daerah: String,
}

#[tokio::main]
async fn main() -> anyhow::Result<()> {
    let input = std::env::var("JAIS_DATA").unwrap_or_else(|_| "legacy/data/jais-petaling.json".into());
    let raw = std::fs::read_to_string(&input)?;
    let records: Vec<JaisRecord> = serde_json::from_str(&raw)?;
    tracing_subscriber::fmt().with_env_filter("info").init();
    dotenvy::dotenv().ok();
    let url = std::env::var("DATABASE_URL")?;
    let pool = db::connect(&url).await?;
    db::apply_schema(&pool).await?;
    let mut inserted = 0u64;
    for r in &records {
        let res = sqlx::query(
            "INSERT INTO quest_mosques (name, address, district, jais_id) VALUES ($1, $2, $3, $4)
             ON CONFLICT (jais_id) DO NOTHING",
        )
        .bind(&r.nama)
        .bind(&r.alamat)
        .bind(&r.daerah)
        .bind(r.jaisId.to_string())
        .execute(&pool)
        .await?;
        inserted += res.rows_affected();
    }
    println!("done: {inserted} quest mosques inserted");
    Ok(())
}
